### PR TITLE
Initial commit for plugin

### PR DESCRIPTION
**Pull request by vthapar**
Adds skeletal plugin
README on how to compile

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -3,9 +3,46 @@ module github.com/submariner-io/coastguard
...
+	github.com/xiang90/probing v0.0.0-20190116061207-43a291ad63a2 // indirect
+	go.etcd.io/bbolt v1.3.3 // indirect
+	go.uber.org/atomic v1.4.0 // indirect
+	go.uber.org/multierr v1.1.0 // indirect
+	go.uber.org/zap v1.10.0 // indirect
+	google.golang.org/grpc v1.22.0 // indirect
+	gopkg.in/jcmturner/goidentity.v3 v3.0.0 // indirect
 )
 
 replace github.com/bronze1man/goStrongswanVici => github.com/mangelajo/goStrongswanVici v0.0.0-20190223031456-9a5ae4453bd
```
Comment (tpantelis) [inline go.mod]: Is this needed, ie lighthouse doesn't use strongSwan?

File: go.mod [hunk 2]
```diff
@@ -3,9 +3,46 @@ module github.com/submariner-io/coastguard
 go 1.12
```
Comment (tpantelis) [inline go.mod]: The module above should reference lighthouse.

File: plugin/lighthouse.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+	// Use a non-writer so we don't write NXDOMAIN to client
+	nw := nonwriter.New(w)
+
+	// Call next plugin (if any).
+	return plugin.NextOrFailure(lh.Name(), lh.Next, ctx, nw, r)
+}
+
+// Name implements the Handler interface.
+func (lh Lighthouse) Name() string { return "lighthouse" }
```
Comment (tpantelis) [inline plugin/lighthouse.go]: The return and end brace should be on separate lines.

File: plugin/setup.go [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+	if c.NextArg() {
+		// If there was another token, return an error, because we currently don't have any configuration.
+		// Any errors returned from this setup function should be wrapped with plugin.Error, so we
+		// can present a slightly nicer error message to the user.
+		return plugin.Error("lighthouse", c.ArgErr())
+	}
+
+	// Add the Plugin to CoreDNS, so Servers can use it in their plugin chain.
+	dnsserver.GetConfig(c).AddPlugin(func(next plugin.Handler) plugin.Handler {
+		return Lighthouse{Next: next}
```
Comment (tpantelis) [inline plugin/setup.go]: I think you want to return `& Lighthouse(...)`

File: go.mod (hunk 1 above)
Comment (mangelajo) [reply go.mod]: True, this can be removed.

File: plugin/README.md [hunk 5]
```diff
@@ -0,0 +1,51 @@
+# lighthouse
+
+## Name
+
+*Lighthouse* - serve services across clusters.
```
Comment (mangelajo) [inline plugin/README.md]: serve services? :-)
 
what about DNS discovery for services across clusters?

File: plugin/README.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+kubernetes:kubernetes
+lighthouse:github.com/submariner-io/lighthouse/plugin/lighthouse
+~~~
+* `make -f Makefile.release DOCKER=your-docker-repo release`
+* `make -f Makefile.release DOCKER=your-docker-repo docker`
+* `make -f Makefile.release DOCKER=your-docker-repo docker-push`
+
+## Examples
+
+~~~ txt
```
Comment (mangelajo) [inline plugin/README.md]: use "`" instead of ~    ?

File: plugin/README.md [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+. {
+    errors
+    log
+    kubernetes cluster.local {
+      fallthrough
+    }
+    lighthouse
+}
+~~~
```
Comment (mangelajo) [inline plugin/README.md]: same here

File: plugin/setup_test.go [hunk 8]
```diff
@@ -0,0 +1,21 @@
+package lighthouse
```
Comment (mangelajo) [inline plugin/setup_test.go]: you need to adapt to use ginkgo/gomega.

File: plugin/setup.go (hunk 4 above)
Comment (vthapar) [reply plugin/setup.go]: Used this as reference:
https://github.com/coredns/example/blob/master/setup.go#L41

File: plugin/README.md (hunk 5 above)
Comment (vthapar) [reply plugin/README.md]: Changed to discover, wording was from template README I picked up.

File: plugin/README.md (hunk 6 above)
Comment (vthapar) [reply plugin/README.md]: It will become a multiline block as we figure out and add more options.

File: go.mod (hunk 1 above)
Comment (vthapar) [reply go.mod]: done

File: plugin/lighthouse.go (hunk 3 above)
Comment (vthapar) [reply plugin/lighthouse.go]: Done

File: plugin/README.md (hunk 7 above)
Comment (vthapar) [reply plugin/README.md]: Done

File: plugin/setup_test.go (hunk 8 above)
Comment (vthapar) [reply plugin/setup_test.go]: Will do in subsequent patches. Still figuring out how to write tests with ginkgo

File: plugin/setup.go (hunk 4 above)
Comment (mangelajo) [reply plugin/setup.go]: Yes but we agreed to use ginkgo, which is just a different framework. I believe you can map the same thing to ginkgo.

Comment (vthapar) [reply plugin/setup.go]: This reply looks out of place here. did you mean this for the setup_test.go? If yes, I'll be making the switch to ginkgo in subsequent patch, if okay with you.

**Pull request merged**